File: SIH/cyber-risk-platform/frontend/src/types/telemetry.ts
export type TelemetrySeverity = 'informational' | 'low' | 'medium' | 'high' | 'critical';

export interface TelemetryEvent {
  id: string;
  organization_id: string;
  asset_id: string | null;
  source: string;
  event_type: string;
  severity: TelemetrySeverity;
  message: string | null;
  source_event_id: string | null;
  occurred_at: string;
  created_at: string;
  event_data: Record<string, any>;
}

export interface PaginatedTelemetry {
  items: TelemetryEvent[];
  page: number;
  page_size: number;
  total: number;
  total_pages: number;
}

export interface TelemetryStats {
  total_events: number;
  critical_events: number;
  high_events: number;
  medium_events: number;
  low_events: number;
  informational_events: number;
  by_source: Record<string, number>;
  by_event_type: Record<string, number>;
}

export interface TelemetryEventFilters {
  organization_id?: string;
  asset_id?: string;
  source?: string;
  event_type?: string;
  severity?: string;
  search?: string;
  from_time?: string;
  to_time?: string;
  page?: number;
  page_size?: number;
  sort_by?: string;
  sort_order?: 'asc' | 'desc';
}

File: SIH/cyber-risk-platform/backend/app/schemas/telemetry.py
import uuid
from pydantic import BaseModel, ConfigDict
from datetime import datetime
from typing import Any

class TelemetryEventRead(BaseModel):
    id: uuid.UUID
    organization_id: uuid.UUID
    asset_id: uuid.UUID | None = None
    source: str
    event_type: str
    severity: str
    message: str | None = None
    event_data: dict[str, Any]
    source_event_id: str | None = None
    occurred_at: datetime
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)
